src/Core/Queues/IQueue.cs

```diff
@@ -21,10 +21,16 @@ public interface IQueue<T> : IHaveSerializer, IDisposable where T : class {
 
         Task<QueueEntry<T>> DequeueAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        [Obsolete("Use QueueEntry<T> overload")]

```

Comment:
Yes, thank you very much! I am going to try and remember why we changed the interfaces to just send the string ids. If I remember right it was just because we have seen other queues that wouldn't be able to implement the interface with the full object. I think it might be ServiceBus queues. Also, we did some stuff in QueueBase that could be helpful to you:

https://github.com/exceptionless/Foundatio/blob/master/src/Core/Queues/QueueBase.cs#L93